src/lib/pages/frontsite/Documents/DocumentDetails/DocumentMetadata/DocumentMetadataExtensions.js

```diff
@@ -0,0 +1,43 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Divider, Table } from 'semantic-ui-react';
+import _merge from 'lodash/merge';
+import _keys from 'lodash/keys';
+import _get from 'lodash/get';
+import _pickBy from 'lodash/pickBy';
+import { extensionsConfig } from '@config';
+
+export const DocumentMetadataExtensions = ({ extensions }) => {
+  extensions = _pickBy(extensions, (value, key) => {
+    return _get(extensionsConfig.fields[key], 'isVisible', true);
+  });
+
+  let result = {};
+
+  _keys(extensions).map(key => {
+    const [objName, objProp] = key.split(':');
```

Comment:
I am not 100% sure I understand, but is this splitting done to avoid the `:`? Is the field name that we send to the backend `accelerator_experiments:accelerator` or only `accelerator`? Or why this split?
Would it be possible to make this manipulation a little bit more readable/understandable?


I would avoid any manipulation to display a label, as I was commenting this is what is done in `invenio-search-js/ui` and it really does not work. What we should do instead is to add a `label` field in the config so that the user can customize what name to display.

Reply:
The convention used in the backend for extensions is `{'<namespace>:<field>': <value> }`, so our data look like

```js
extensions: {
  accelerator_experiments:accelerator: 'AccelerateMe',
  accelerator_experiments:project: 'CDS',
}
```
and so on. 

So here I am transforming the object to the following form to avoid rendering repetitive information.  
```
extensions: {
  accelerator_experiments: {
    accelerator: 'AccelerateMe',
    project: 'CDS',
  }
}
```
I think that moving it in the `api/document/serializer` and also deserialize when we submit DocumentForm is cleaner and will hold the conversion in one place only.

Dropping some of my thoughts, then this will not match with our ui field configuration structure, which if it gets changed will not match with the backend field configuration structure.

@ntarocco wdyt?